src/ReactiveSocket.cpp

```diff
@@ -75,6 +76,20 @@ void ReactiveSocket::requestSubscription(
   automaton->start();
 }
 
+void ReactiveSocket::requestFireAndForget(
```

Comment:
I would be inclined to keep the callback (Subscriber or otherwise), precisely because you often don't care about the result.  With a callback the method could be setup to return immediately and invoke the callback, client can choose to ignore it.  But without the callback we have to wait for it to actually succeed or fail to be written.